<launch>
  <arg name="cam_width" default="1280" />
  <arg name="cam_height" default="960" />
  <arg name="cam_model" default="daA1280-54uc" />
  <arg name="camera_frame" default="camera" />
  
  <arg name="cam_image_topic" default="/camera/image_raw" />
  <arg name="cam_info_topic" default="/camera/camera_info" />
  
  <arg name="marker_size" default="1.5" />

  <!-- Choose the calibration file that corresponds to the resolution being used. -->
  <arg name="cam_calibration_file" default="file://$(find autopic)/config/$(arg cam_model)_$(arg cam_width)x$(arg cam_height).yaml" />

  <node name="static_tf" pkg="tf" type="static_transform_publisher" args="0 -0.02 0.07 0 0 1 0 tip_triangle camera 100"/>

  <node name="basler_camera" pkg="basler_camera" type="basler_camera_node" output="screen">
    <param name="frame_rate" type="int" value="10" />
    <param name="frame_id" type="string" value="$(arg camera_frame)" />
    <param name="camera_info_url" type="string" value="$(arg cam_calibration_file)" />
  </node>

  <node ns="camera" pkg="image_proc" type="image_proc" name="image_proc"/>
</launch>
